src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalTestDetectors.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Infrastructure.Common
+{
+    // This class contains helper "detector" methods to detect specific
+    // conditions about the environment in which the tests are run.
+    // They are used for [ConditionalFact] and [ConditionalTheory]
+    // tests.  They are called at most once, and the return value cached.
+    internal static class ConditionalTestDetectors
+    {
+        private const string ClientCertificateSubject = "WCF Client Certificate";
+        private const string CertificateIssuer = "DO_NOT_TRUST_WcfBridgeRootCA";
+
+        public static bool IsRootCertificateInstalled()
+        {
+            return IsCertificateInStore(StoreName.Root, StoreLocation.LocalMachine, CertificateIssuer) ||
```

Comment:
Reviewers: this is first cut, and feedback is welcome.  Right now the detector only detects and does not attempt to install the certificates.  We also have a duplication of issue name literals that we should find a way to share.  These changes are not expected to light up ToF or Helix tests currently using [ConditionalFact] on the new detectors.  It is necessary to find ways to install the certificates prior to running those tests.